config/proxy/proxy.go

```diff
@@ -0,0 +1,197 @@
+// Description: This file contains the implementation of the new consolidated proxy/geolocation config.
+
+package proxyconfig
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/getlantern/eventual/v2"
+	"github.com/getlantern/golog"
+	"github.com/getlantern/rot13"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/getlantern/flashlight/v7/apipb"
+)
+
+const (
+	defaultConfigSaveDir  = ""
+	defaultConfigFilename = "proxies.conf"
+)
+
+// alias for better readability
+type ProxyConfig = apipb.ConfigResponse
```

Comment:
Yeah, I know. I originally was going to use `ClientConfig` but that's already used in `config/client_config.go` and I couldn't think of anything else at the moment. I figured once we include the global config we'd be able to rename it to just `Config` since it wouldn't clash. 

In case you haven't noticed by now, I'm not so great at naming :laughing:. So I'm open to any and all suggestions.

Reply:
`ProxyConfig` has been renamed to `UserConfig`, as well as package name, etc., since it stores the user's location, IP, and the proxies assigned to them. Maybe, one day, we'll be able to use `config`..